lib/web/apiserver.go

```diff
@@ -2693,6 +2696,61 @@ func (h *Handler) WithClusterAuth(fn ClusterHandler) httprouter.Handle {
 	})
 }
 
+// ProvisionTokenHandler is a authenticated handler that is called for some existing Token
+type ProvisionTokenHandler func(w http.ResponseWriter, r *http.Request, p httprouter.Params, site reversetunnel.RemoteSite, roles types.SystemRoles) (interface{}, error)
```

Comment:
Why not just pass the entire token to the handler instead of roles? Would make things a little cleaner I think.